src/java.base/share/classes/java/lang/ClassValue.java

```diff
@@ -302,43 +315,50 @@ static class Identity {
     private volatile Version<T> version = new Version<>(this);
     Version<T> version() { return version; }
     void bumpVersion() { version = new Version<>(this); }
-    static class Version<T> {
+    static final class Version<T> {
```

Comment:
Since this is all final components, it might serve better as a record?

Reply:
Note that this class has a significant identity. Does that make it still suitable for record conversion? I think we need a comment on its identity significance if we convert.